README.md

```diff
@@ -54,9 +54,87 @@ We encourage auditors to monitor this log, and aim to help people access the dat
 A simple example would be a service that emails users (on a different address) when ceritficates have been issued on their behalf.
 This can then be used to detect bad behavior or possible compromise.
 
-## Parameters
+## CA / KMS support
 
-The fulcio root CA is currently running on GCP Private CA with the EC_P384_SHA384 algorithm.
+### Google Cloud Platform Private CA
+
+The public fulcio root CA is currently running on GCP Private CA with the EC_P384_SHA384 algorithm.
+
+You can also run with your own GCP Private CA, by passing in a parent and google as the CA
+
+```
+go run main.go serve --ca googleca  --gcp_private_ca_parent=projects/myproject/locations/us-central1/certificateAuthorities/myproject
+```
+
+### SoftHSM (pkcs11)
+
+> :warning: A SoftHSM does not provide the same security guarantees as hardware based HSM
+> Use for test development purposes only.
+
+fulcio may also be used with a pkcs11 SoftHSM. You will also need `pkcs11-tool`
+
+
+Create a `config/crypto11.conf` file::
+
+```
+{
+"Path" : "/usr/lib64/softhsm/libsofthsm.so",
+"TokenLabel": "fulcio",
+"Pin" : "2324"
+}
+```
+
+And a `config/softhsm2.conf`
+
+```
+directories.tokendir = /tmp/tokens
+objectstore.backend = file
+log.level = INFO
+```
+
+Export the `config/softhsm2.conf`
+
+```
+export SOFTHSM2_CONF=`pwd`/config/softhsm2.cfg 
+```
+
+### Start a SoftHSM instance
+
+```
+softhsm2-util --init-token --slot 0 --label fulcio
+```
+
+### Create keys within the SoftHSM
+
+```
+pkcs11-tool --module /usr/lib64/softhsm/libsofthsm.so --login --login-type user --keypairgen --id 1 --label FulcioCA  --key-type EC:secp384r1
+```
+
+* Note: you can import existing keys and import using pkcs11-tool, see pkcs11-tool manual for details
+
+### Create a root CA
+
+Now that your keys are generated, you can use the fulcio `createca` command to generate a Root CA. This command
+will also store the generated Root CA into the HSM by the delegated id passed to `--hsm-caroot-id`
+
+```
+fulcio createca--org=acme --country=UK --locality=SomeTown --province=SomeProvince --postal-code=XXXX --street-address=XXXX --hsm-caroot-id 99 --out myrootCA.pem
+```
+
+### Run fulcio with the SoftHSM
+
+```
+fulcio serve --ca fulcioca --hsm-caroot-id 99
```

Comment:
should we name this `softhsm` instead of `fulcioca`?

Reply:
The thinking is that this is a pkcs11 implementation and so it should work with HW HSM's as well (the SoftHSM specifics are in the config file). So this becomes fulcioca (as it creates and signs the client certificates in much the same manner as googleca does).